packages/demo-wallet-client/src/ducks/sep24DepositAsset.ts

```diff
@@ -51,6 +59,16 @@ export const depositAssetAction = createAsyncThunk<
       throw new Error("Something went wrong, home domain is not defined.");
     }
 
+    // This is unlikely
+    if (
+      custodialIsEnabled &&
+      !(custodialSecretKey && custodialPublicKey && custodialMemoId)
+    ) {
+      throw new Error(
+        "Custodial mode requires secret key, public key, and memo ID",
+      );
+    }
```

Comment:
Is it possible? Doesn't the custodial component ensure these values are collected before initiating a deposit?

Reply:
This should never happen, adding here just in case something goes wrong when updating the state.